modules/server/src/test/scala/observe/server/InstrumentControllerSimSpec.scala

```diff
@@ -51,120 +37,100 @@ class InstrumentControllerSimSpec extends CatsSuite {
-      _ <- tio.sleep(tick) // give it enough time for at least one tick
+      _ <- IO.sleep(tick) // give it enough time for at least one tick
       _ <- sim.stopObserve
-      r <- f.join
-    } yield r).unsafeRunSync()
-    assert(result === ObserveCommandResult.Stopped)
+      r <- f.joinWithNever
+    } yield assertEquals(r, ObserveCommandResult.Stopped)
   }
   test("pause/stop pause observation") {
-    implicit val tio = IO.timer(ExecutionContext.global)
-    val sim          = simulator
-    val result       = (for {
+    val sim = simulator
+    for {
       f <- sim.observe(toImageFileId("S001"), 900.milliseconds).start
-      _ <- tio.sleep(tick) // give it enough time for at least one tick
+      _ <- IO.sleep(tick) // give it enough time for at least one tick
       _ <- sim.pauseObserve
-      _ <- tio.sleep(tick) // give it enough time for at least one tick
+      _ <- IO.sleep(tick) // give it enough time for at least one tick
       r <- sim.stopPaused
-      _ <- f.join
-    } yield r).unsafeRunSync()
-
-    assert(result === ObserveCommandResult.Stopped)
+      _ <- f.joinWithNever
+    } yield assertEquals(r, ObserveCommandResult.Stopped)
   }
   test("pause/resume observation") {
-    implicit val tio = IO.timer(ExecutionContext.global)
-    val sim          = simulator
-    val result       = (for {
+    val sim = simulator
+    for {
       f <- sim.observe(toImageFileId("S001"), 900.milliseconds).start
-      _ <- tio.sleep(tick) // give it enough time for at least one tick
+      _ <- IO.sleep(tick) // give it enough time for at least one tick
       _ <- sim.pauseObserve
       r <- sim.resumePaused
-      _ <- f.join
-    } yield r).unsafeRunSync()
-
-    assert(result === ObserveCommandResult.Success)
+      _ <- f.joinWithNever
+    } yield assertEquals(r, ObserveCommandResult.Success)
   }
   test("pause/stop observation") {
-    implicit val tio = IO.timer(ExecutionContext.global)
-    val sim          = simulator
-    val result       = (for {
+    val sim = simulator
+    for {
       f <- sim.observe(toImageFileId("S001"), 2.seconds).start
-      _ <- tio.sleep(tick) // give it enough time for at least one tick
+      _ <- IO.sleep(tick) // give it enough time for at least one tick
       _ <- sim.pauseObserve
-      _ <- tio.sleep(tick) // give it enough time for at least one tick
+      _ <- IO.sleep(tick) // give it enough time for at least one tick
       _ <- sim.stopObserve
-      _ <- tio.sleep(tick) // give it enough time for at least one tick
-      r <- f.join
-    } yield r).unsafeRunSync()
+      _ <- IO.sleep(tick) // give it enough time for at least one tick
+      r <- f.joinWithNever
+    } yield assertEquals(r, ObserveCommandResult.Paused)
 
     // FIXME The simulator should return Stopped
-    // assert(result === ObserveCommandResult.Stopped)
-    assert(result === ObserveCommandResult.Paused)
+    // assertEquals(r, ObserveCommandResult.Stopped)
```

Comment:
Commented code

Reply:
But, that commented code is relevant to the FIXME